src/psyclone/tests/psyir/symbols/containersymbol_test.py

```diff
@@ -0,0 +1,142 @@
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors R. W. Ford, A. R. Porter and S. Siso, STFC Daresbury Lab
+#         I. Kavcic, Met Office
+#         J. Henrichs, Bureau of Meteorology
+# -----------------------------------------------------------------------------
+
+''' Perform py.test tests on the psygen.psyir.symbols.containersymbols file '''
+
+import pytest
+from psyclone.psyir.symbols import ContainerSymbol, SymbolError
+from psyclone.psyir.symbols.containersymbol import ContainerSymbolInterface, \
+    FortranModuleInterface
+from psyclone.psyGen import Container
+from psyclone.configuration import Config
+
+
+def test_containersymbol_initialisation():
+    '''Test that a ContainerSymbol instance can be created when valid
+    arguments are given, otherwise raise relevant exceptions.'''
+
+    sym = ContainerSymbol("my_mod")
+    assert isinstance(sym, ContainerSymbol)
+    assert sym.name == "my_mod"
+    assert not sym._reference  # Reference are not evaluated until told
+    # Right now the FortranModuleInterface is assigned by default
+    # because it is the only one. This may change in the future
+    assert sym._interface == FortranModuleInterface
+
+    with pytest.raises(TypeError) as error:
+        sym = ContainerSymbol(None)
+    assert "ContainerSymbol name attribute should be of type 'str'" \
+        in str(error)
+
+
+def test_containersymbol_str():
+    '''Test that a ContainerSymbol instance can be stringified'''
+
+    sym = ContainerSymbol("my_mod")
+    assert str(sym) == "my_mod: <not linked>"
+
+    sym._reference = Container("my_mod")
+    assert str(sym) == "my_mod: <linked>"
+
+
+def test_containersymbol_resolve_external_container(monkeypatch):
+
+    sym = ContainerSymbol("my_mod")
+
+    monkeypatch.setattr(sym._interface, "import_container",
+                        staticmethod(lambda x: "MockContainer"))
+
+    # At the beggining the reference is never resolved (lazy evaluation)
+    assert not sym._reference
+
+    # When container is invoked the reference is resolved
+    assert sym.container == "MockContainer"
+    assert sym._reference == "MockContainer"
+
+    # Following invokations do not update the container reference
+    monkeypatch.setattr(sym._interface, "import_container",
+                        staticmethod(lambda x: "OtherContainer"))
+    assert sym.container == "MockContainer"
+
+
+def test_containersymbol_generic_interface():
+    '''Check ContainerSymbolInterface abstract methods '''
+
+    abstractinterface = ContainerSymbolInterface
```

Comment:
`ContainerSymbolInterface` and its sub-classes do not have state and just have a staticmethod. So it seemed more appropriate to use the class instead of an instance for now. Note that inside ContainerSymbol it is also used like this.

Reply:
OK, thanks for explaining.